client/src/routes/Session/components/ContentControls/ContentControls.tsx

```diff
@@ -105,12 +114,17 @@ const ContentControls: React.FC<ContentControlsProps> = ({
         variant="tertiary"
         disabled={!exercise.slide.next}
         RightIcon={ChevronRight}
-        onPress={() =>
+        onPress={() => {
+          if (isLoadingContent || !exercise.slide.next) {
```

Comment:
This check is necessary cause the disabling of the button could be delayed and more press events could happen after the slide was actually changed on the UI.